include/aos/common/crypto/openssl/opensslprovider.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_OPENSSL_PROVIDER_HPP_
+#define AOS_OPENSSL_PROVIDER_HPP_
+
+#include <aos/common/config.hpp>
+#include <aos/common/crypto/crypto.hpp>
+
+#define AOS_SIGNER             "aossigner"
+#define AOS_ALGORITHM          "AOS"
+#define PKEY_PARAM_AOS_KEYPAIR AOS_ALGORITHM "PrivateKey"
+
+#define OPENSSL_ERROR()                                                                                                \
```

Comment:
Don't use macro like function, not recomended by misra. Use just function instead.